File: src/Array/python/PascalTriangle-2.py
__author__ = 'haoxiang'
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        res = [0]*(rowIndex+1)
        if rowIndex == 0:
            res =[[1]]
            return res[0]
        if rowIndex == 1:
            res = [[1],[1,1]]
            return res[1]
        else :
            res[0] = [1]
            res[1] = [1,1]
            for i in range(2,rowIndex+1):
                res[i] = [0]*(i+1)
                res[i][0] = res[i][i] = 1
                for j in range(1,i):
                    res[i][j] = res[i-1][j-1]+res[i-1][j]
            print res[rowIndex]
            return res[rowIndex]

Test = Solution()
Test.getRow(5)

File: src/Array/cpp/qiuyi/PascalTriangle-2.cpp
class Solution {
public:
	vector<int> getRow(int rowIndex) {
		vector<vector<int>> triangle;

		for (int i = 0; i < rowIndex; i++) {
			vector<int> temp;
			for (int j = 0; j <= i; j++) {
				if (j == 0 || j == i)
					temp.push_back(1);
				else
					temp.push_back(triangle[i-1][j-1]+triangle[i-1][j]);
			}
			triangle.push_back(temp);
		}
		return triangle[rowIndex];
	}
};